ts/Extensions/DataGrouping/DataGroupingSeriesComposition.ts

```diff
@@ -202,73 +203,82 @@ function anchorPoints(
-            let firstGroupstEnd;
 
-            if (isNumber(groupStart) && isNumber(groupLength)) {
-                firstGroupstEnd = groupStart + (groupLength - 1);
-            }
+    if (!(
+        dataGroupingOptions &&
+            series.xData &&
+            totalRange &&
+            series.groupMap
+    )) {
+        return;
+    }
 
-            groupedXData[0] = ({
-                middle: groupedXData[0] + 0.5 * totalRange,
-                end: groupedXData[0] + totalRange,
-                firstPoint: series.xData[0],
-                lastPoint: firstGroupstEnd && series.xData[firstGroupstEnd]
-            } as AnchorChoiceType)[firstAnchor];
+    const groupedDataLastIndex = groupedXData.length - 1,
+        anchor = dataGroupingOptions.anchor,
+        firstAnchor = dataGroupingOptions.firstAnchor,
+        lastAnchor = dataGroupingOptions.lastAnchor;
+    let i = groupedXData.length - 1,
+        iEnd = 0;
+    // Anchor points that are not extremes.
+
+    // Change the first point position, but only when it is
+    // the first point in the data set not in the current zoom.
+    if (firstAnchor && series.xData[0] >= groupedXData[0]) {
+        iEnd++;
+        const groupStart = series.groupMap[0].start,
+            groupLength = series.groupMap[0].length;
+        let firstGroupEnd;
+
+        if (isNumber(groupStart) && isNumber(groupLength)) {
+            firstGroupEnd = groupStart + (groupLength - 1);
         }
 
-        // Change the last point position but only when it is
-        // the last point in the data set not in the current zoom.
-        if (
+        groupedXData[0] = ({
+            start: groupedXData[0],
+            middle: groupedXData[0] + 0.5 * totalRange,
+            end: groupedXData[0] + totalRange,
+            firstPoint: series.xData[0],
+            lastPoint: firstGroupEnd && series.xData[firstGroupEnd]
+        } as AnchorChoiceType)[firstAnchor];
+    }
+
+    // Change the last point position but only when it is
+    // the last point in the data set not in the current zoom,
+    // or if it is not the 1st point simutainously.
+    if (
+        groupedDataLastIndex > 0 &&
             lastAnchor &&
-            lastAnchor !== 'start' &&
             totalRange &&
-            groupedXData[groupedDataLength] >= xMax - totalRange
-        ) {
-            const lastGroupStart = series.groupMap[
-                series.groupMap.length - 1
-            ].start;
-
-            groupedXData[groupedDataLength] = ({
-                middle: groupedXData[groupedDataLength] + 0.5 * totalRange,
-                end: groupedXData[groupedDataLength] + totalRange,
-                firstPoint: lastGroupStart && series.xData[lastGroupStart],
-                lastPoint: series.xData[series.xData.length - 1]
-            } as AnchorChoiceType)[lastAnchor];
+            groupedXData[groupedDataLastIndex] >= xMax - totalRange
+    ) {
+        i--;
```

Comment:
I suggested a more descriptive name for `i` variable. It is easy to miss it and it is not obvious why you decrease the iterator here. Previously `i` was used close to its assignment so it was more readable